src/NzbDrone.Core/Download/Clients/rTorrent/RTorrent.cs

```diff
@@ -0,0 +1,178 @@
+            var remainingTries = 5;
+            var retryDelay = 500; //ms
+            var ready = false;
+            while (remainingTries-- > 0)
+            {
+                if (ready = _proxy.IsHashReadyTorrent(hash, Settings)) break;
+                else Thread.Sleep(retryDelay);
+            }
+           
+            if (!ready)
+            {
+                RemoveItem(hash, false);
+
+                return null;
+            }
+            else
+            {
+                _proxy.SetTorrentLabel(hash, Settings.TvCategory, Settings);
+
+                var priority = (RTorrentPriority)(remoteEpisode.IsRecentEpisode() ?
+                                        Settings.RecentTvPriority : Settings.OlderTvPriority);
+                _proxy.SetTorrentPriority(hash, Settings, priority);
+
+                return hash;
+            }
+        }
+
+        protected override String AddFromTorrentFile(RemoteEpisode remoteEpisode, String hash, String filename, Byte[] fileContent)
+        {
+            _proxy.AddTorrentFromFile(filename, fileContent, Settings);
+
+            _proxy.SetTorrentLabel(hash, Settings.TvCategory, Settings);
+
+            var priority = (RTorrentPriority)(remoteEpisode.IsRecentEpisode() ?
+                                    Settings.RecentTvPriority : Settings.OlderTvPriority);
+            _proxy.SetTorrentPriority(hash, Settings, priority);
+
+            return hash;
+        }
+
+        public override IEnumerable<DownloadClientItem> GetItems()
+        {
+            try
+            {
+                var torrents = _proxy.GetTorrents(Settings);
+
+                var items = new List<DownloadClientItem>();
+                foreach (RTorrentTorrent torrent in torrents)
+                {
+                    var item = new DownloadClientItem();
+                    item.DownloadClient = Definition.Name;
+                    item.Title = torrent.Name;
+                    item.DownloadId = torrent.Hash;
+                    item.OutputPath = _remotePathMappingService.RemapRemoteToLocal(Settings.Host, new OsPath(torrent.BaseDir));
+                    item.TotalSize = torrent.TotalSize;
+                    item.RemainingSize = torrent.RemainingSize;
+                    if (torrent.RemainingTime != -1) item.RemainingTime = TimeSpan.FromSeconds(torrent.RemainingTime);
+                    item.Category = torrent.Category;
+
+                    if (torrent.IsFinished) item.Status = DownloadItemStatus.Completed;
+                    else if (torrent.IsActive) item.Status = DownloadItemStatus.Downloading;
+                    else if (!torrent.IsActive) item.Status = DownloadItemStatus.Paused;
+                    else if (!torrent.IsOpen) item.Status = DownloadItemStatus.Queued;
+
+                    if (item.Status != DownloadItemStatus.Completed) item.IsReadOnly = true;
+                    // FIXME Set conditions for  isreadonly = false 
```

Comment:
I'm uncertain what you mean by delete here. If it is queued to seed, wouldn't the user want to keep it in the client and not have it deleted?